tests/subsys/net/openthread/rpc/client/src/main.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <mock_nrf_rpc_transport.h>
+
+#include <zephyr/ztest.h>
+#include <zephyr/kernel.h>
+
+#include <openthread/ip6.h>
+
+#include <ot_rpc_ids.h>
+
+/* CBOR constants */
+
+#define CBOR_FALSE 0xf4
+#define CBOR_TRUE  0xf5
+
+/* Macros for constructing nRF RPC packets for the OpenThread command group. */
+
+#define RPC_PKT(bytes...)                                                                          \
+	(mock_nrf_rpc_pkt_t)                                                                       \
+	{                                                                                          \
+		.data = (uint8_t[]){bytes}, .len = sizeof((uint8_t[]){bytes}),                     \
+	}
+
+#define RPC_INIT_REQ RPC_PKT(0x04, 0x00, 0xff, 0x00, 0xff, 0x00, 'o', 't')
+#define RPC_INIT_RSP RPC_PKT(0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 'o', 't')
+
+#define RPC_CMD(cmd, ...) RPC_PKT(0x80, cmd, 0xff, 0x00, 0x00 __VA_OPT__(, ) __VA_ARGS__, 0xf6)
+#define RPC_RSP(...)	  RPC_PKT(0x01, 0xff, 0x00, 0x00, 0x00 __VA_OPT__(, ) __VA_ARGS__, 0xf6)
+
+static void nrf_rpc_err_handler(const struct nrf_rpc_err_report *report)
+{
+	zassert_ok(report->code);
+}
+
+static void tc_setup(void *f)
+{
+	mock_nrf_rpc_tr_expect_add(RPC_INIT_REQ, RPC_INIT_RSP);
+	zassert_ok(nrf_rpc_init(nrf_rpc_err_handler));
+	mock_nrf_rpc_tr_expect_reset();
+}
+
+ZTEST(ot_rpc_client, test_otIp6SetEnabled)
+{
+	otError error;
+
+	/* Test successful set to false */
```

Comment:
I'd recommend splitting it on 3 test cases. Even it seems simple enough. It will help determine bug easier when we accumulate a lot of test.

Reply:
ok, will do